mitodl/open-discussions
# Add management command to index comments and posts

#### What are the relevant tickets?
Fixes #641 

#### What's this PR do?
Adds a management command to iterate over posts and comments that the user is subscribed to. 

#### How should this be manually tested?
First, delete the elasticsearch docker container and recreate it.

Add a new environment variable `INDEXING_API_USERNAME` with a username which is subscribed to the channels you wish to index (you may want to use `mitodl` which is the default user used to index from micromasters).

Run `./manage.py index_content`. You should see a celery task id being output, then the management command should wait. Run `sudo reddit-restart` in the reddit shell to temporarily interrupt reddit, to assert the autoretry functionality. (Make sure to run `docker-compose build` beforehand, the latest version of celery is required.)

Before it is done, go to `localhost:9101/_alias` and verify that there is an alias ending with `_reindexing` and none ending with `_default`. After the reindex completes go to that URL again. You should not see any alias ending with `_reindexing` but there should be one ending with `_default`.

Go to `localhost:9101/_search`. You should see many documents.
